src/tests/EShopworld.WorkerProcess.IntegrationTests/DistributedLockTests.cs

```diff
@@ -0,0 +1,112 @@
+﻿using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.IO;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Eshopworld.Core;
+using Eshopworld.DevOps;
+using Eshopworld.Telemetry;
+using Eshopworld.Tests.Core;
+using EShopworld.WorkerProcess.Configuration;
+using EShopworld.WorkerProcess.CosmosDistributedLock;
+using FluentAssertions;
+using Microsoft.ApplicationInsights;
+using Microsoft.Azure.Documents;
+using Microsoft.Azure.Documents.Client;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.DependencyInjection.Extensions;
+using Microsoft.Extensions.Options;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace EShopworld.WorkerProcess.IntegrationTests
+{
+    public class DistributedLockTests
+    {
+        private readonly ServiceProvider _serviceProvider;
+        private readonly ITestOutputHelper _output;
+
+        public DistributedLockTests(ITestOutputHelper output)
+        {
+            var configuration = new ConfigurationBuilder()
+                .SetBasePath(Directory.GetCurrentDirectory())
+                .AddJsonFile("appsettings.json", false, true)
+                .Build();
+
+            var serviceCollection = new ServiceCollection();
+            ConfigureServices(serviceCollection, configuration);
+
+            _serviceProvider = serviceCollection.BuildServiceProvider();
+
+            _output = output;
+        }
+
+        [Fact, IsIntegration]
+        public void AcquireLock_ForNewLock_ShouldNotThrowException()
+        {
+            // Arrange
+            Func<Task> act = async () =>
+            {
+                var distributedLock = _serviceProvider.GetService<IDistributedLock>();
+                using (await distributedLock.Acquire("test1"))
+                {
+                    // create a recurring job
```

Comment:
the comment is not valid :)

Reply:
I see that was there just to indicate the usage of code. I'll change it.

Reply:
yes, i know :) i just think this will not be recurring job that people will write here so maybe something like:
`// do critical-section code here` or whatever else bit more generic :)